R/guides-.r

```diff
@@ -208,6 +208,11 @@ guides_merge <- function(gdefs) {
 
 # process layer information
 guides_geom <- function(gdefs, layers, default_mapping) {
+  # remove layers with show.legend=FALSE or NA
```

Comment:
Are you sure this logic is correct? `NA` means use the default, i.e. show if the aesthetic is used